core/git_mixins/active_branch.py

```diff
@@ -34,7 +35,8 @@ def get_branch_status(self):
         if first_line.startswith("## Initial commit on "):
             return "Initial commit on `{}`.".format(first_line[21:])
 
-        short_status_pattern = r"## ([A-Za-z0-9\-_\/]+)(\.\.\.([A-Za-z0-9\-_\/]+)( \[((ahead (\d+))(, )?)?(behind (\d+))?\])?)?"
+        branch_pattern = "[A-Za-z0-9" + re.escape(re.sub("[~\\^\\:\\?\\*\\[\"]", "", string.punctuation)) + "]+?"
```

Comment:
Without the `?`, it greedily gobbles up remotes, too. So in the case of

    ## master...origin/master

With the `?` it yields `master` as the branch name, with `origin/master` as the remote name.
Without the `?` it yields `master...origin/master` as the branch name, no remote recognized.